Mathlib/Algebra/Category/ModuleCat/Injective.lean

```diff
@@ -3,36 +3,64 @@ Copyright (c) 2023 Jujian Zhang. All rights reserved.
 Released under Apache 2.0 license as described in the file LICENSE.
 Authors: Jujian Zhang
 -/
-import Mathlib.Algebra.Category.ModuleCat.ChangeOfRings
-import Mathlib.Algebra.Category.Grp.EnoughInjectives
-import Mathlib.Algebra.Category.Grp.ZModuleEquivalence
+import Mathlib.Algebra.Module.Injective
+import Mathlib.GroupTheory.Divisible
+import Mathlib.Algebra.EuclideanDomain.Int
 import Mathlib.Logic.Equiv.TransferInstance
+import Mathlib.RingTheory.PrincipalIdealDomain
+import Mathlib.CategoryTheory.Preadditive.Injective
+import Mathlib.Algebra.Category.ModuleCat.EpiMono
 
 /-!
-# Category of $R$-modules has enough injectives
-
-we lift enough injectives of abelian groups to arbitrary $R$-modules by adjoint functors
-`restrictScalars ⊣ coextendScalars`
-
-## Implementation notes
-This file is not part of `Algebra/Module/Injective.lean` to prevent import loop: enough-injectives
-of abelian groups needs `Algebra/Module/Injective.lean` and this file needs enough-injectives of
-abelian groups.
+# Injective objects in the category of $R$-modules
 -/
 
 open CategoryTheory
 
-universe v u
-variable (R : Type u) [Ring R]
+universe u v
+variable (R : Type u) (M : Type v) [Ring R] [AddCommGroup M] [Module R M]
+namespace Module
+
+theorem Baer.of_divisible [DivisibleBy M ℤ] : Module.Baer ℤ M := fun I g ↦ by
+  rcases IsPrincipalIdealRing.principal I with ⟨m, rfl⟩
+  obtain rfl | h0 := eq_or_ne m 0
+  · refine ⟨0, fun n hn ↦ ?_⟩
+    rw [Submodule.span_zero_singleton] at hn
+    subst hn
+    exact (map_zero g).symm
+  let gₘ := g ⟨m, Submodule.subset_span (Set.mem_singleton _)⟩
+  refine ⟨LinearMap.toSpanSingleton ℤ M (DivisibleBy.div gₘ m), fun n hn ↦ ?_⟩
+  rcases Submodule.mem_span_singleton.mp hn with ⟨n, rfl⟩
+  rw [map_zsmul, LinearMap.toSpanSingleton_apply, DivisibleBy.div_cancel gₘ h0, ← map_zsmul g,
+    SetLike.mk_smul_mk]
```

Comment:
I think this shouldn't be moved, and then this file will need very few imports.

Reply:
I don't think the number of imports in this file matters all that much, but reverted anyway since I don't care much about the imports within category theory itself.